src/Microsoft.OData.Client/BatchSaveResult.cs

```diff
@@ -753,7 +773,11 @@ private IEnumerable<OperationResponse> HandleBatchResponse(ODataBatchReader batc
                         #endregion
 
                         default:
-                            Error.ThrowBatchExpectedResponse(InternalError.UnexpectedBatchState);
+                            // In ODataJsonLightBatchReader, readerState remains Initial after calling Read() the first time
+                            if (!this.useJsonBatch)
+                            {
+                                Error.ThrowBatchExpectedResponse(InternalError.UnexpectedBatchState);
+                            }

```

Comment:
Really? This seems like a bug in the ODataJsonLightBatchReader -- your state shouldn't depend on the format.

Reply:
@mikepizzo Definitely a bug in `ODataJsonLightBatchReader`